vendor/assets/javascripts/trumbowyg/vendor/plugins/contextbar/trumbowyg.contextbar.js

```diff
@@ -87,6 +88,14 @@
                     rect.width == 0;
                   };
 
+                  var getSaveBoundingClientRect = function(node) {
+                    if (node["getBoundingClientRect"]) {
+                      return node.getBoundingClientRect();
+                    } else {
+                      return getSaveBoundingClientRect(node.parentNode);
+                    };
```

Comment:
What do you think if to add a terminal condition? 
```
if (!node.parentNode) { return }
```
Theoretically it shouldn't go into infinite loop because would stop at the top of a tree, but

Reply:
this comment is really hard to understand for not directly involved dev's.

Reply:
> this comment is really hard to understand for not directly involved dev's.

The issue is when browser try to call `getBoundingClientRect` func, for some reason it doesn't have this method. To fix this behaviour Hauke asked for a method call on parental node. So theoretically recursion can reach the toppest node of a page. But Hauke already added a terminal condition to the func.

Reply:
ok, thanks for you explanation.